service/k8sclient/kubectl/kubectl_test.go

```diff
@@ -19,12 +19,60 @@ package kubectl
 
 import (
 	"context"
+	"crypto/sha256"
+	"io/ioutil"
+	"os"
+	"strings"
 	"testing"
 
 	"github.com/stretchr/testify/assert"
 	"github.com/stretchr/testify/require"
+
+	"github.com/percona-platform/dbaas-controller/utils/app"
 )
 
+func TestNewKubeCtl(t *testing.T) {
+	ctx := app.Context()
+
+	cmd, err := getKubectlCmd(ctx)
+	require.NoError(t, err)
+
+	validKubeconfig, err := run(ctx, cmd, []string{"config", "view", "-o", "json"}, nil)
+	require.NoError(t, err)
+
+	t.Run("BasicNewKubeCtl", func(t *testing.T) {
+		sha256KubeconfigExpected := sha256.Sum256(validKubeconfig)
+		kubeCtl, err := NewKubeCtl(ctx, string(validKubeconfig))
+		require.NoError(t, err)
+		// lookup for kubeconfig path
+		var kubeconfigFlag string
+		for _, option := range kubeCtl.cmd {
+			if strings.HasPrefix(option, "--kubeconfig") {
+				kubeconfigFlag = option
+				break
+			}
+		}
+
+		assert.True(t, strings.HasSuffix(kubeconfigFlag, kubeconfigFileName))
+
+		kubeconfigFilePath := strings.Split(kubeconfigFlag, "=")[1]
+		dat, err := ioutil.ReadFile(kubeconfigFilePath) //nolint:gosec
+		require.NoError(t, err)
+		sha256KubeconfigActual := sha256.Sum256(dat)
+		assert.Equal(t, sha256KubeconfigExpected, sha256KubeconfigActual)
```

Comment:
why do we compare hash sum instead of body of config?

Reply:
Had the intention to reuse it in other test cases. But, here we have only one. Agree, following YAGNI - I'll remove it.